File: Core/Models/Idiom.swift
//
//  Idiom.swift
//  My Dictionary
//
//  Created by Aleksandr Riakhin on 3/8/25.
//

import Foundation

struct Idiom: Identifiable, Hashable {
    let id: String
    var idiom: String
    var definition: String
    let timestamp: Date
    var examples: [String]
    var isFavorite: Bool

    init(
        idiom: String,
        definition: String,
        id: String,
        timestamp: Date,
        examples: [String],
        isFavorite: Bool
    ) {
        self.definition = definition
        self.id = id
        self.idiom = idiom
        self.isFavorite = isFavorite
        self.timestamp = timestamp
        self.examples = examples
    }
}

File: MyDictionaryAndroid/app/src/main/java/com/dor/mydictionary/core/Idiom.kt
package com.dor.mydictionary.core

import java.util.Date

data class Idiom(
    val id: String,
    val idiomItself: String,
    val definition: String,
    val timestamp: Date,
    val isFavorite: Boolean,
    val examples: List<String>,
    val difficultyScore: Int = 0,
    val languageCode: String = "en"
) {
    val difficultyLevel: Difficulty
        get() = Difficulty.fromScore(difficultyScore)
    
    val difficultyLabel: String
        get() = difficultyLevel.displayName
    
    val shouldShowDifficultyLabel: Boolean
        get() = difficultyScore > 0
} 